src/llm/wmt_model.py

```diff
@@ -382,20 +443,65 @@ def forward(
         self,
         encoder_input: Float[Tensor, "batch enc_seq_len"],  # type: ignore
         decoder_input: Float[Tensor, "batch dec_seq_len"],  # type: ignore
+        pad_id: int = 1,
     ) -> Float[Tensor, "batch dec_seq_len vocab_tgt_size"]:  # type: ignore
+        ############# ENCODER #############
+
+        # calculate the attention mask for the encoder and decoder
+        encoder_attention_mask = encoder_input != pad_id
```

Comment:
I think we should delegate this task to data loading pipeline.

From the `transformers` perspective, we have a `tokenizer` which transforms raw text into `intputs`. `inputs` here is a culmination of `input_ids` and `attention_masks`.

WDYT?

Reply:
yes you are right. that would be more efficient. I did it here in this script because I needed to debug it properly. We will move it to the collate_fn.

Reply:
Cool -- after you debug this bit, let's move this to the data pipeline.

Reply:
Moved it to the data pipeline.